src/ui/actions/layout.ts

```diff
@@ -41,13 +41,20 @@ export function toggleCommandPalette(): UIThunkAction {
   };
 }
 export function setViewMode(viewMode: ViewMode): UIThunkAction {
-  return async ({ dispatch }) => {
+  return async ({ dispatch, getState }) => {
     // There's a possible race condition here so it's important to handle the nag logic first.
     // Otherwise, it's possible for the nag to not be properly dismissed.
     if (viewMode === "dev" && !(await isLocalNagDismissed(LocalNag.YANK_TO_SOURCE))) {
       await dismissLocalNag(LocalNag.YANK_TO_SOURCE);
       dispatch(setSelectedPrimaryPanel("explorer"));
     }
+    // If switching to non-dev mode, we check the selectedPrimaryPanel and update to comments
+    // if selectedPrimaryPanel is one that should only be visible in dev mode.
+    const selectedPrimaryPanel = getSelectedPrimaryPanel(getState());
+    const viewerPanels = ["comments", "events"]
```

Comment:
Can we move this array to `src/ui/state/layout.ts`? Something like:

```diff
diff --git a/src/ui/state/layout.ts b/src/ui/state/layout.ts
index e572f7662..61ab0a4f7 100644
--- a/src/ui/state/layout.ts
+++ b/src/ui/state/layout.ts
@@ -8,6 +8,8 @@ export type LayoutState = {
 };
 
 export type ViewMode = "dev" | "non-dev";
-export type PrimaryPanelName = "explorer" | "debugger" | "comments" | "events" | "search";
+export const VIEWER_PANELS = ["events", "comments"] as const;
+type ViewerPrimaryPanelName = typeof VIEWER_PANELS[number];
+export type PrimaryPanelName = "explorer" | "debugger" | "search" | ViewerPrimaryPanelName;
 export type SecondaryPanelName = "console" | "inspector" | "network" | "react-components";
 export type PanelName = PrimaryPanelName | SecondaryPanelName;
```

Reply:
Updated via https://github.com/RecordReplay/devtools/pull/5203/commits/2ccff53cdb1baa77ce406efb3c14ef7c45a8cf25 but I had to set the type of `selectedPrimaryPanel` to `any` when using the `includes()` array otherwise TS was mad. I was looking at https://github.com/microsoft/TypeScript/issues/26255 and there didn't seem to be a clear alternative to using `any`, but let me know if there's a better way! Thanks!